In [13]:
import numpy as np
import matplotlib.pyplot as plt

import fsps

## UV Star Formation Rates

## Part (a)

In [14]:
### constants
L_sun = 3.846 * 10**33 # erg/s
pc = 3.085677581467192 * 10**18 # cm
m_ab0 = -2.5 * np.log10(3631 * 10**(-23)) # AB magnitude zero point
h = 6.6260755 * 10**(-27) # erg s
c = 2.99792458 * 10**18 # angstrom/s

In [16]:
sp = fsps.StellarPopulation(compute_vega_mags=False, sfh=1, const=1, zmet=10, imf_type=2, sf_trunc=13.7)

In [17]:
sp.params['tage'] = 13.7 # Gyr

In [18]:
SFR = sp.sfr
print(SFR)

7.29927007299e-11


In [19]:
M_ab = sp.get_mags(tage=13.7, bands=['galex_fuv'])
M_ab[0]

6.8068425521015783

Equation 12 of Kennicutt & Evans (2012) says that $\log_{10}(\dot{M_*}) = \log_{10}({\rm L_{FUV}}) - \log_{10}({\rm C_{FUV}})$, where ${\rm L_{FUV}} = \nu_{\rm FUV} \times L_{\rm FUV}$.

Below are the series of steps to take us from an FUV-band AB magnitude to ${\rm L_{FUV}}$ as required by equation 12.

$f_{\rm FUV} = 10^{-0.4*(m_{\rm FUV} + m_{ab,0})}$

$L_{\rm FUV} = 4 * \pi * 10^2 * f_{\rm FUV}$. We evaluate at a distance of 10pc because there the apparent and absolute magnitudes have the same value.

${\rm L_{FUV}}$ $=$ $L_{\rm FUV} \times \nu_{\rm FUV}$.




In [20]:
# get filter
fuv = fsps.get_filter('galex_fuv')
# get wavelength (in Angstroms)
fuv.lambda_eff

1535.1

In [21]:
# compute frequency
fuv_freq = c / fuv.lambda_eff
fuv_freq

1952918103055175.8

In [22]:
# area correspoding to d=10pc
A = 4 * np.pi * (10*pc)**2
A

1.1964951828447585e+40

In [23]:
# flux
f_fuv = 10**(-0.4*(M_ab[0]+m_ab0))
f_fuv

6.8752616092531728e-23

In [24]:
# luminosity as needed by Kennicut eqn 12
L_fuv = fuv_freq * A * f_fuv
L_fuv

1.6065128872840995e+33

In [25]:
# now to get the calibration
C_fuv = L_fuv/SFR
print(C_fuv)
print(np.log10(C_fuv))

2.20092265558e+43
43.3426047809


## Part (b)

My value of $\log {\rm C}_{\rm FUV}$ is 43.34. This agrees with the Kennicutt & Evans (2012) value of 43.35.